tests/components/modbus/test_light.py

```diff
@@ -314,9 +338,101 @@ async def test_light_service_turn(
                 {
                     CONF_NAME: TEST_ENTITY_NAME,
                     CONF_ADDRESS: 1234,
+                    CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                    CONF_SCAN_INTERVAL: 0,
+                    CONF_BRIGHTNESS_REGISTER: 1,
+                    CONF_COLOR_TEMP_REGISTER: 2,
+                },
+                {
+                    CONF_NAME: f"{TEST_ENTITY_NAME} 2",
+                    CONF_ADDRESS: 1234,
+                    CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                    CONF_SCAN_INTERVAL: 0,
+                    CONF_BRIGHTNESS_REGISTER: 1,
+                },
+            ]
+        }
+    ],
+)
+async def test_color_temp_light(
+    hass: HomeAssistant, mock_modbus_ha, modbus_light_entity
+) -> None:
+    """Test Modbus Light color temperature amd brightness."""
+    assert hass.states.get(ENTITY_ID).state == STATE_OFF
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_ON,
+        service_data={
+            ATTR_ENTITY_ID: ENTITY_ID,
+            ATTR_BRIGHTNESS: 128,
+            ATTR_COLOR_TEMP_KELVIN: 2000,
+        },
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get(ENTITY_ID).state == STATE_ON
+    calls = mock_modbus_ha.write_register.call_args_list
+    expected_modbus_brightness = 50
+    expected_modbus_temp = 0
+    assert any(
+        call.args[0] == 1 and call.kwargs["value"] == expected_modbus_brightness
+        for call in calls
+    )
+    assert any(
+        call.args[0] == 2 and call.kwargs["value"] == expected_modbus_temp
+        for call in calls
+    )
+    color_temp = 2000
+    brightness = 255
+    modbus_color_temp = 0
+    incorrect_input = "invalid"
+    result = await modbus_light_entity.async_set_brightness(incorrect_input)
+    assert result is None
+    modbus_light_entity._brightness_address = None
```

Comment:
this should be part of the test parameterization, so that the test runs once for a light with brightness_address, and another for one without